Lib/gftools/builder/__init__.py

```diff
@@ -281,7 +285,22 @@ def run_fontmake(self, source, args):
             original_output_dir = args["output_dir"]
             tmpdir = tempfile.TemporaryDirectory()
             args["output_dir"] = tmpdir.name
-        args["flatten_components"] = self.config["flattenComponents"]
+
+        if (
+            self.config["flattenComponents"] or
+            self.config["decomposeTransformedComponents"]
+        ):
+            filters = args.get("filters", [])
+            if self.config["flattenComponents"]:
+                filters.append([
+                    loadFilterFromString("FlattenComponentsFilter")
+                ])
```

Comment:
```suggestion
            if self.config["flattenComponents"]:
                filters.append(
                    loadFilterFromString("FlattenComponentsFilter")
                )
```

You're appending a list which contains a single filter. What you want is to just append the filter.